### PR TITLE
rc 2.5.18

### PR DESCRIPTION
**Pull request by spond**

### CODE REVIEW

File: res/TemplateBatchFiles/SelectionAnalyses/aBSREL.bf [hunk 1]
```diff
@@ -496,6 +501,24 @@ selection.io.json_store_branch_attribute(absrel.json, terms.json.rate_distributi
                                                       absrel.branch.rate_distributions);
 
 
+absrel.full_model.fit [terms.likelihood_function] = absrel.likelihood_function_id;
```
Comment (stevenweaver) [inline res/TemplateBatchFiles/SelectionAnalyses/aBSREL.bf:504]: Pertains to #1198

File: res/TemplateBatchFiles/libv3/UtilityFunctions.bf [hunk 2]
```diff
@@ -1033,22 +1033,20 @@ lfunction utility.CatersianProduct (arguments) {
...
-{
-	for (_gb_idx2 = 0; _gb_idx2 < theValue; _gb_idx2=_gb_idx2+1)
-	{
+function _sortStringsAux (theKey, theValue) {
+    
+ 	for (_gb_idx2 = 0; _gb_idx2 < theValue; _gb_idx2 += 1) {
 		_gb_sortedStrings [_gb_idx] = theKey;
-		_gb_idx = _gb_idx + 1;
+		_gb_idx += 1;
+		console.log (_gb_idx);
```
Comment (stevenweaver) [inline res/TemplateBatchFiles/libv3/UtilityFunctions.bf:1041]: stray console.log

File: res/TemplateBatchFiles/libv3/models/codon/BS_REL.bf [hunk 3]
```diff
@@ -358,8 +359,14 @@ function models.codon.BS_REL.post_definition(model) {
 function models.codon.BS_REL.get_branch_length(model, tree, node) {
 	parameters.SetLocalModelParameters (model, tree, node);
 	parameters.SetCategoryVariables   (model);
-	bl = Eval (model [utility.getGlobalValue("terms.model.branch_length_string")]) / 3;
-	//console.log ("Branch length = " + bl);
+  bl = utility.GetEnvVariable ("BRANCH_LENGTH_STENCIL");
```
Comment (stevenweaver) [inline res/TemplateBatchFiles/libv3/models/codon/BS_REL.bf:362]: formatting

File: res/TemplateBatchFiles/libv3/models/model_functions.bf [hunk 4]
```diff
@@ -617,11 +617,22 @@ lfunction model.BranchLengthExpressionFromMatrix (q,freqs,is_canonical) {
 	by_expr = {};
 	dim = Rows (q);
 	can_alias = Type (freqs[0]) == "Number";
+  stencil = utility.GetEnvVariable ("BRANCH_LENGTH_STENCIL");
```
Comment (stevenweaver) [inline res/TemplateBatchFiles/libv3/models/model_functions.bf:620]: formatting

File: res/TemplateBatchFiles/libv3/tasks/trees.bf [hunk 5]
```diff
@@ -766,6 +766,63 @@ lfunction trees.ConjunctionLabel (tree_id, given_labels) {
 
 
 
+   return {"labeled" : inodes_labeled, "labels" : resulting_labels};
+   // pass 1 to choose the best state for subtree parents
+}
+
+/**
```
Comment (stevenweaver) [inline res/TemplateBatchFiles/libv3/tasks/trees.bf:773]: cool function

File: run_unit_tests.sh [hunk 6]
```diff
@@ -5,6 +5,8 @@
 #export ASAN_OPTIONS=detect_leaks=1
 
 currentDir=$PWD
+#export ASAN_OPTIONS=detect_leaks=1
```
Comment (stevenweaver) [inline run_unit_tests.sh:8]: Address Sanitizer option - https://clang.llvm.org/docs/AddressSanitizer.html

Comment (stevenweaver) [reply run_unit_tests.sh:8]: Should be part of the cmake build as opposed to in shell script

File: src/core/associative_list.cpp [hunk 7]
```diff
@@ -164,12 +164,18 @@ void _AssociativeList::Duplicate (BaseRefConst br) {
 
 //_____________________________________________________________________________________________
 
-HBLObjectRef _AssociativeList::MCoord (HBLObjectRef p) {
+HBLObjectRef _AssociativeList::MCoord (HBLObjectRef p, HBLObjectRef cache) {
```
Comment (stevenweaver) [inline src/core/associative_list.cpp:167]: Neat, reminds me of lru_cache option in python.

File: src/core/batchlanruntime.cpp [hunk 8]
```diff
@@ -565,7 +565,7 @@ bool      _ElementaryCommand::HandleGetDataInfo (_ExecutionList& current_program
 
                             for (long species_index = filter_source->NumberSpecies()-1; species_index >= 0; species_index --) {
                                 filter_source->RetrieveState(site,species_index,*buffer, false);
-                                filter_source->Translate2Frequencies (*buffer, storage->theData,  count_gaps >= 0.5);
+                                filter_source->Translate2Frequencies (*buffer, storage->theData,  count_gaps);
```
Comment (stevenweaver) [inline src/core/batchlanruntime.cpp:568]: Will always correct_for_exclusions.

File: CMakeLists.txt [hunk 9]
```diff
@@ -407,8 +407,8 @@ if(${MPI_FOUND})
         HYPHYMPI
         PROPERTIES
         if(${OPENMP_FOUND})
-            COMPILE_FLAGS "${DEFAULT_COMPILE_FLAGS} ${DEFAULT_WARNING_FLAGS} ${MPI_COMPILE_FLAGS} ${OpenMP_CXX_FLAGS} "
```
[the remark this answers is not in the document]
Comment (spond) [reply CMakeLists.txt:410]: Done

File: res/TemplateBatchFiles/libv3/UtilityFunctions.bf (hunk 2 above)
Comment (spond) [reply res/TemplateBatchFiles/libv3/UtilityFunctions.bf]: Removed.

File: res/TemplateBatchFiles/libv3/models/codon/BS_REL.bf (hunk 3 above)
Comment (spond) [reply res/TemplateBatchFiles/libv3/models/codon/BS_REL.bf]: Fixed.

File: res/TemplateBatchFiles/libv3/models/model_functions.bf (hunk 4 above)
Comment (spond) [reply res/TemplateBatchFiles/libv3/models/model_functions.bf]: Fixed.

**Pull request merged**